src/ipc-handlers.ts

```diff
@@ -238,26 +236,31 @@ export async function connectWpcomSite(
 
 	const connections = userData.connectedWpcomSites[ currentUserId ];
 
-	sites.forEach( ( siteToAdd ) => {
-		const isAlreadyConnected = connections.some(
-			( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
-		);
+	list.forEach( ( { sites, localSiteId } ) => {
+		sites.forEach( ( siteToAdd ) => {
+			const isAlreadyConnected = connections.some(
+				( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
+			);
 
-		// Add the site if it's not already connected
-		if ( ! isAlreadyConnected ) {
-			connections.push( { ...siteToAdd, localSiteId } );
-		}
+			// Add the site if it's not already connected
+			if ( ! isAlreadyConnected ) {
+				connections.push( { ...siteToAdd, localSiteId } );
+			}
+		} );
 	} );
 
 	await saveUserData( userData );
 
-	return connections.filter( ( conn ) => conn.localSiteId === localSiteId );
+	return connections.filter( ( conn ) =>
+		list.some( ( { localSiteId } ) => conn.localSiteId === localSiteId )
+	);
 }
 
+type WpcomSitesToDisconnect = { siteIds: number[]; localSiteId: string }[];
+
 export async function disconnectWpcomSite(
```

Comment:
```suggestion
export async function disconnectWpcomSites(
```
Nit, but since this function takes an array of IDs, maybe we should rename it?